File: android/src/main/java/com/plugin/flutter/zsdk/SGDParams.java
package com.plugin.flutter.zsdk;

/**
 * Created by luis901101 on 2020-02-10.
 */
public class SGDParams
{
    /** Settings */

    /** Writable settings */
    public static final String KEY_DARKNESS = "print.tone";
    public static final String KEY_PRINT_SPEED = "media.speed";
    public static final String KEY_TEAR_OFF = "ezpl.tear_off";
    public static final String KEY_MEDIA_TYPE = "ezpl.media_type";
    public static final String KEY_PRINT_METHOD = "ezpl.print_method";
    public static final String KEY_PRINT_WIDTH = "ezpl.print_width";
    public static final String KEY_LABEL_LENGTH = "zpl.label_length";
    public static final String KEY_LABEL_LENGTH_MAX = "ezpl.label_length_max";
    public static final String KEY_ZPL_MODE = "zpl.zpl_mode";
    public static final String KEY_POWER_UP_ACTION = "ezpl.power_up_action";
    public static final String KEY_HEAD_CLOSE_ACTION = "ezpl.head_close_action";
    public static final String KEY_LABEL_TOP = "zpl.label_top";
    public static final String KEY_LEFT_POSITION = "zpl.left_position";
    public static final String KEY_PRINT_MODE = "ezpl.print_mode";
    public static final String KEY_REPRINT_MODE = "ezpl.reprint_mode";
    public static final String KEY_PRINTER_LANGUAGES = "device.languages";

    /** Read only settings */
    public static final String KEY_PRINTER_MODEL_NAME = "ip.dhcp.vendor_class_id";
    public static final String KEY_DEVICE_FRIENDLY_NAME = "device.friendly_name";
    public static final String KEY_FIRMWARE = "appl.name";
    public static final String KEY_LINK_OS_VERSION = "appl.link_os_version";
    public static final String KEY_PRINTER_DPI = "head.resolution.in_dpi";

    /** Actions */
    public static final String KEY_MANUAL_CALIBRATION = "ezpl.manual_calibration";

    /** Values */
    public static final String VALUE_ZPL_LANGUAGE = "hybrid_xml_zpl";
    public static final double VALUE_DPI_DEFAULT = 203;
    public static final String VALUE_GET_ALL = "allcv";
}

File: ios/Classes/SGDParams.m
//
//  SGDParams.m
//  zsdk
//
//  Created by Luis on 2/12/20.
//

#import "SGDParams.h"
#import "SGD.h"

@implementation SGDParams
/** Settings */

/** Writable settings */
+ (NSString*)KEY_DARKNESS {return @"print.tone";}
+ (NSString*)KEY_PRINT_SPEED {return @"media.speed";}
+ (NSString*)KEY_TEAR_OFF {return @"ezpl.tear_off";}
+ (NSString*)KEY_MEDIA_TYPE {return @"ezpl.media_type";}
+ (NSString*)KEY_PRINT_METHOD {return @"ezpl.print_method";}
+ (NSString*)KEY_PRINT_WIDTH {return @"ezpl.print_width";}
+ (NSString*)KEY_LABEL_LENGTH {return @"zpl.label_length";}
+ (NSString*)KEY_LABEL_LENGTH_MAX {return @"ezpl.label_length_max";}
+ (NSString*)KEY_ZPL_MODE {return @"zpl.zpl_mode";}
+ (NSString*)KEY_POWER_UP_ACTION {return @"ezpl.power_up_action";}
+ (NSString*)KEY_HEAD_CLOSE_ACTION {return @"ezpl.head_close_action";}
+ (NSString*)KEY_LABEL_TOP {return @"zpl.label_top";}
+ (NSString*)KEY_LEFT_POSITION {return @"zpl.left_position";}
+ (NSString*)KEY_PRINT_MODE {return @"ezpl.print_mode";}
+ (NSString*)KEY_REPRINT_MODE {return @"ezpl.reprint_mode";}
+ (NSString*)KEY_PRINTER_LANGUAGES {return @"device.languages";}

/** Read only settings */
+ (NSString*)KEY_PRINTER_MODEL_NAME {return @"ip.dhcp.vendor_class_id";}
+ (NSString*)KEY_DEVICE_FRIENDLY_NAME {return @"device.friendly_name";}
+ (NSString*)KEY_FIRMWARE {return @"appl.name";}
+ (NSString*)KEY_LINK_OS_VERSION {return @"appl.link_os_version";}
+ (NSString*)KEY_PRINTER_DPI {return @"head.resolution.in_dpi";}
+ (NSString*)KEY_DEVICE_PRINT_HEAD_RESOLUTION {return @"device.printhead.resolution";}

/** Actions */
+ (NSString*)KEY_MANUAL_CALIBRATION {return @"ezpl.manual_calibration";}
+ (NSString*)KEY_VIRTUAL_DEVICE {return @"apl.enable";}

/** Values */
+ (NSString*)VALUE_ZPL_LANGUAGE {return @"hybrid_xml_zpl";}
+ (double)VALUE_DPI_DEFAULT {return 203;}
+ (NSString*)VALUE_GET_ALL {return @"allcv";}
+ (NSString*)VALUE_PDF {return @"pdf";}
+ (NSString*)VALUE_NONE {return @"none";}

@end
